tests/scripts/test_lint_header_self_contained.py

```
@@ -0,0 +1,182 @@
+        self.assertNotIn("/ws/src/ui/src/mainwindow.cpp", command)
+        self.assertIn("-DQT_NO_KEYWORDS", command)
+        self.assertIn("-std=c++17", command)
+        self.assertEqual(command[-4:], ["-fsyntax-only", "-x", "c++", "/tmp/probe.cpp"])
+
+    def test_dedupes_union_against_carried_flags(self):
+        unit = make_unit(
+            "/ws/src/a/a.cpp", "/ws/build", ["-I", "/qt/include", "-I/local"]
+        )
+        command = MODULE.syntax_only_command(
+            unit,
+            pathlib.Path("/tmp/probe.cpp"),
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/probe.cpp"

(S108)